golang/src/config.rs

```diff
@@ -0,0 +1,298 @@
+
+use libc::c_char;
+use pravega_client_shared::*;
+use pravega_client_config::*;
+use pravega_client_config::connection_type::ConnectionType;
+use pravega_client_config::connection_type::MockType::Happy;
+use crate::config::BConnectionType::Tokio;
+use std::ffi::{CStr, CString};
+use std::str::FromStr;
+use std::time::Duration;
+use pravega_client_config::credentials::Credentials;
+use pravega_client_retry::retry_policy::RetryWithBackoff;
+
+
+const SPLIT: &str = ",";
+
+
+#[repr(C)]
+pub struct BStreamConfiguration {
```

Comment:
I rename all the struct names with the `Mapping` suffix, maybe it makes more sense. so the logic will be `Go.StreamConfiguration -> C.StreamConfigurationMapping -> rust.StreamConfiguration`. I think it is quite clear for type conversion between go and rust.
`StreamConfigurationMapping` is also distinguished from `StreamConfiguration`, so people will not get confused when they review the binding-related codes.

Reply:
I fixed some of them. The rest of them are something like "variant is never constructed: `Keycloak`", which means it's never used in this rust project, but it's necessary for generating the C types so that the golang can use it.